### PR TITLE
Session 19

### PR DESCRIPTION
**Pull request by javarb**
Hello this is the session about reaching HTTP endpoints from tests

### CODE REVIEW

File: sessions/2018/december/session-19.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+## Session 19 (04/12/2018)
+
+- Testing to HTTP endpoints
+- Review past homeworks
+
+### Notes:
+
+#### Testing HTTP endpoints
+
+We cloned the 'Talenhaven' Roger's repository and from it he was explaining to me all concepts related with reach HTTP endpoint from tests.
```
Comment (kinbiko) [inline sessions/2018/december/session-19.md]: Let's leave the name of the repo out as it's not public.

File: sessions/2018/december/session-19.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+His approach was to buit a generic test to reach HTTP endpoints regardless the HTTP verb used.
+
+#### Homework revision
+
+Is recommended that the JS fronted shows results in text, this can be achieved using some library but the most simple thing to do is to return an JSON object in the API that contains a text with the calculated value and then use that string in the frontend.
```
Comment (kinbiko) [inline sessions/2018/december/session-19.md]: What do you mean by `results`? This sentence is very ambiguous just now, and from reading it I can't remember what it was about.

File: sessions/2018/december/session-19.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+**TIP:** Some popular frontend libraries that have all embeeded are React and Angular. As they have corresponding learning curves, [Create React App][1] can be useful.
+
+by other side, about the `{}` being prefereable than `[]` at the beggining of a JSON file, besides of the security reason, another reason to prefeer `{}` is because if we need to change or add something to our response will be easier if we have an object instead an array. If we have an array, and want to return another thing, we'll have make a new object and add that new object to the JSON payload anyway.
```
Comment (kinbiko) [inline sessions/2018/december/session-19.md]: The key point is about not breaking backwards compatibility.
If you have to change from `[]` to `{}` all your clients using the old API will break, or you'll have to maintain both implementations on the server.
On the other hand, if you use `{}` from the beginning, you can add an array as part of this, pretty easily by making it a property of that object.

File: sessions/2018/december/session-19.md (hunk 2 above)
Comment (javarb) [reply sessions/2018/december/session-19.md]: I meant the API response values, I have updated it.

**Pull request merged**